### PR TITLE
RN-1170: Matrix UI updates

### PR DESCRIPTION
**Pull request by alexd-bes**
### Changes:
- Updated styling as per new design, including applying to chart and map overlay tables (confirmed with Felicity)
- Updated theme to include customised theme variables in the `palette` object and updated the `types` to match as per [MUI docs](https://v4.mui.com/guides/typescript/#customization-of-theme)

---

#### Standard matrix
Screen Shot 2024-02-26 at 3 14 10 PM

#### Nested matrix rows
Screen Shot 2024-02-26 at 3 12 40 PM

#### Updated chart table styling
Screen Shot 2024-02-26 at 3 13 34 PM

#### Updated map overlay table styling
Screen Shot 2024-02-26 at 3 14 31 PM

### CODE REVIEW

File: packages/tupaia-web/src/features/Map/Map.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ const StyledMap = styled(LeafletMap)`
     top: -50px;
     right: 3px;
     a {
-      background: ${({ theme }) => theme.navigationBtn.main};
+      background: #34353f;
```
Comment (alexd-bes) [inline packages/tupaia-web/src/features/Map/Map.tsx:64]: This theme var was used in just this place, so it made sense to leave it contained in the component

File: packages/tupaia-web/src/features/Visuals/Matrix/Matrix.tsx [hunk 2]
```diff
@@ -21,23 +21,37 @@ import { DashboardItemContext } from '../../DashboardItem';
...
+    padding-left: 0.7rem;
   }
-`;
-
-const Wrapper = styled.div`
-  overflow: hidden;
-  max-height: clamp(
-    20rem,
-    70vh,
-    60rem
```
Comment (alexd-bes) [inline packages/tupaia-web/src/features/Visuals/Matrix/Matrix.tsx:37]: These settings were moved into `theme.ts`

File: packages/ui-components/src/components/DataTable/StyledTable.tsx [hunk 3]
```diff
@@ -5,58 +5,22 @@
 import styled from 'styled-components';
 import MuiTable from '@material-ui/core/Table';
 
-const DARK_THEME_BORDER = 'rgb(82, 82, 88)';
-
 export const StyledTable = styled(MuiTable)`
   table-layout: fixed;
-  overflow: hidden;
-  border: 1px solid ${({ theme }) => (theme.palette.type === 'light' ? 'none' : DARK_THEME_BORDER)};
```
Comment (alexd-bes) [inline packages/ui-components/src/components/DataTable/StyledTable.tsx:10]: These styles were moved out of `ui-components` and into the theme of the package that uses it, so it's easy to change these wherever we want to

File: packages/ui-components/src/components/Matrix/MatrixRow.tsx [hunk 4]
```diff
@@ -194,10 +255,22 @@ export const MatrixRow = ({ row, parents = [] }: MatrixRowProps) => {
...
+  const getClassNames = () => {
+    const highlightedClass = isExpanded || depth > 0 ? 'highlighted' : '';
+    const matrixClass = 'matrix';
+    const childClass = depth > 0 ? 'child' : 'parent';
+
+    const baseClass = `${matrixClass} ${highlightedClass} ${childClass}`;
+    if (depth > 0 || index === undefined) return baseClass;
+    if (index % 2 === 0) return `${baseClass} even`;
+    return `${baseClass} odd`;
+  };
```
Comment (alexd-bes) [inline packages/ui-components/src/components/Matrix/MatrixRow.tsx:267]: Class names here seemed like a better way to handle things, and to be able to set colours in `tupaia-web` as well, as opposed to props

File: packages/tupaia-web/src/features/Map/MapOverlaySelector/MobileMapOverlaySelector.tsx [hunk 5]
```diff
@@ -32,11 +32,11 @@ const ExpandButton = styled(Button)`
   align-items: center;
   border-radius: 0;
   padding: 1rem;
-  background-color: ${({ theme }) => theme.overlaySelector.menuBackground};
+  background-color: ${({ theme }) => theme.palette.overlaySelector.menuBackground};
   position: relative;
   &:hover,
   &:focus {
```
Comment (jaskfla) [inline packages/tupaia-web/src/features/Map/MapOverlaySelector/MobileMapOverlaySelector.tsx:38]:
```suggestion
  &:focus,
  &:focus-visible {
```

File: packages/tupaia-web/src/theme/theme.ts [hunk 6]
```diff
@@ -1,99 +1,138 @@
...
+    head: {
+      backgroundColor: theme.palette.table?.header,
+    },
+  },
+  MuiTableCell: {
+    root: {
+      backgroundColor: 'inherit',
+      fontSize: '0.875rem',
+      paddingBlock: '0.8rem',
+      paddingLinline: '1.56rem',
```
Comment (jaskfla) [inline packages/tupaia-web/src/theme/theme.ts:123]: Is this a typo? 🧐 

```suggestion
      paddingInline: '1.56rem',
```

File: packages/ui-components/src/components/Matrix/Matrix.tsx [hunk 7]
```diff
@@ -63,9 +60,8 @@ export const Matrix = ({ columns = [], rows = [], disableExpand, ...config }: Ma
           <Table component={MatrixTable} ref={tableEl} stickyHeader>
             <MatrixHeader />
             <TableBody>
-              {rows.map(row => (
-                // add a random key to avoid bugs with re-rendering
```
Comment (jaskfla) [inline packages/ui-components/src/components/Matrix/Matrix.tsx:67]: 🍾

File: packages/ui-components/src/components/Matrix/MatrixCell.tsx [hunk 8]
```diff
@@ -89,16 +85,24 @@ export const MatrixCell = ({ value, rowTitle, isCategory, colKey, onClick }: Mat
...
+        <Dot
+          $color={presentation?.color}
+          aria-label={`${presentation?.description ? `${presentation.description}: ` : ''}${
+            value || 'No value'
+          }`}
+        />
+      );
+    }
+    if (value === null || value === undefined) {
+      return '-';
```
Comment (jaskfla) [inline packages/ui-components/src/components/Matrix/MatrixCell.tsx:101]:
```suggestion
      return '—'; // em dash
```

File: packages/ui-components/src/components/Matrix/MatrixCell.tsx [hunk 9]
```diff
@@ -89,16 +85,24 @@ export const MatrixCell = ({ value, rowTitle, isCategory, colKey, onClick }: Mat
...
-    value
-  );
+
+  const getDisplayValue = () => {
+    if (isDots) {
+      return (
+        <Dot
+          $color={presentation?.color}
+          aria-label={`${presentation?.description ? `${presentation.description}: ` : ''}${
+            value || 'No value'
```
Comment (jaskfla) [inline packages/ui-components/src/components/Matrix/MatrixCell.tsx:95]: Minor, but I personally found this expression a bit of a mind-bender

```suggestion
          aria-label={!!presentation?.description ? `${presentation.description}: ${value}` : 'No value'}
```

**Review by jaskfla — APPROVED** [review]
Some neat refactors along the way! Left some minor nitpicks, but approving 🎉

**Pull request merged**